src/ql/arch/cc/pass/gen/vq1asm/detail/codegen.cc

```diff
@@ -544,250 +724,432 @@ void Codegen::customGate(
+                // NB: value NO_STATIC_CODEWORD_OVERRIDE (-1) means 'no override'
+                bi.staticCodewordOverride = Settings::findStaticCodewordOverride(instruction, csv.operandIdx, iname);
 #endif
             } else if (bi.signalValue == csv.signalValueString) {           // signal unchanged
                 // do nothing
             } else {
                 showCodeSoFar();
-                QL_FATAL(
+                QL_USER_ERROR(
                     "Signal conflict on instrument='" << csv.si.ic.ii.instrumentName
                     << "', group=" << csv.si.group
                     << ", between '" << bi.signalValue
                     << "' and '" << csv.signalValueString << "'"
-                );  // FIXME: add offending instruction
+                );
             }
         }
 
-        // store signal duration
+        // store some attributes
         bi.durationInCycles = durationInCycles;
+        bi.isMeasure = csv.isMeasure;
+        bi.describe = ir::describe(custom);
+
+        // handle measurement (for which we'll generate output allowing downstream software to retrieve them)
+        if (csv.isMeasure) {
+            checkMeasOps();
+            bi.measQubit = ops.qubits[0];
+        }
 
-#if OPT_FEEDBACK
-        // FIXME: assumes that group configuration for readout input matches that of output
-        // store operands used for readout, actual work is postponed to bundleFinish()
-        if (isReadout) {
+        // store operands used for real-time measurements, actual work is postponed to bundle_finish()
+        if (settings.isMeasRsltRealTime(*custom.instruction_type)) {
+            // FIXME: move the checks to collectCodeGenInfo?
+            // FIXME: at the output side, similar checks are not performed
             /*
-             * kernel->gate allows 3 types of measurement:
+             * In the old IR, kernel->gate allows 3 types of measurement:
              *         - no explicit result. Historically this implies either:
              *             - no result, measurement results are often read offline from the readout device (mostly the raw values
              *             instead of the binary result), without the control device ever taking notice of the value
              *             - implicit bit result for qubit, e.g. for the CC-light using conditional gates
              *         - creg result (old, no longer valid)
              *             note that Creg's are managed through a class, whereas bregs are just numbers
              *         - breg result (new)
+             *
+             *  In the new IR (or, better said, in the new way "prototype"s for instruction operands van be defined
+             *  using access modes as described in
+             *  https://openql.readthedocs.io/en/latest/gen/reference_configuration.html#instructions-section
```

Comment:
Is this a new issue? Of what? Because it is the intention to support this in general. And then codegen could add constraints on the actual bit locations that get updated by a measurement.

Reply:
I don't think it's an issue for now, it will become so if we want to do fancy feedback stuff.

It's a result of the available access modes, specifically "M", which states:
_M for a measurement of the qubit for which the result is written to its implicitly associated bit register (DDG W for the qubit and its bit, liveness use for the qubit, and liveness kill for the bit); and`_

Equivalent access modes that allow explicit specifications for a qubit and a bit separately do not seem to exist

Reply:
The implicit association bothers me. Requiring implementing this for each target is wrong. I know there is some backward compatibility here and also cQASM compatibility stuff. But those latter can be get rid off. I would like to take a step back here and investigate without taking care of those backward and cQASM compatibility issues.

Reply:
Many things are 'wrong' with the way "measure" is currently defined. First of all, the implicit result bit, which was in the original language spec, also see https://libqasm.readthedocs.io/en/latest/cq1-instructions.html#default-instruction-set. On the one hand it does not properly allow assigning to some explicit bit (one would like to be able to write `b[23] = measure(q[0])`, on the other hand it does not allow qubits (that are part of the qubit register) to be measured without storing the bit result.
Then there is the assumption that a measurement produces binary data, and not e.g. 1 of 3 possibilities. And then we have calibration, where measurements are performed that produce (e.g.) raw IQ values.

My proposal would be to create an issue for this, as it is out of scope of the current PR

Reply:
Agree.